internal/command/deploy/machines.go

```diff
@@ -383,12 +386,31 @@ func (md *machineDeployment) latestImage(ctx context.Context) (string, error) {
 	return resp.App.CurrentReleaseUnprocessed.ImageRef, nil
 }
 
-func (md *machineDeployment) setMachineGuest(vmSize string) error {
-	if vmSize == "" {
-		return nil
-	}
+func (md *machineDeployment) setMachineGuest(vmSize string, vmCPUKind string, vmCPUs int, vmMem int) error {
 	md.machineGuest = &api.MachineGuest{}
```

Comment:
I think we have to start from a known baseline default like `shared-cpu-1x`  (that is what `nil` does)

Add just after this line the following code:
```go
md.machineGuest.SetSize("shared-cpu-1x")
```
 
and then only try to override size, cpus, kind and memory if the flags are not the empty values.